lib/services/service_brokers/v2/catalog_plan.rb

```diff
@@ -37,10 +46,13 @@ def validate!
       validate_hash!(:metadata, metadata) if metadata
       validate_bool!(:free, free) if free
       validate_bool!(:bindable, bindable) if bindable
+      validate_hash!(:schemas, @schemas_data) if @schemas_data
     end
 
     def validate_schemas!
-      errors.add_nested(schemas, schemas.errors) unless schemas.valid?
+      if schemas
+        errors.add_nested(schemas, schemas.errors) unless schemas.valid?
+      end
```

Comment:
I would simplify this as

    if schemas && !schemas.valid?  
      errors.add_nested(schemas, schemas.errors)
    end